raster-foundry/raster-foundry
# Click to select analysis and edit histogram, input parameters

## Overview
Allow clicking an analysis in the project analyses view to adjust color maps on a histogram and view more details about the parameter types.

### Checklist

- ~Styleguide updated, if necessary~
- ~[Swagger specification](https://github.com/raster-foundry/raster-foundry-api-spec) updated~
- ~Symlinks from new migrations present or corrected for any new migrations~
- ~Any content changes are properly templated using `BUILDCONFIG.APP_NAME`~
- ~Any new SQL strings have tests~

### Demo
![image](https://user-images.githubusercontent.com/4392704/55079651-fa755c80-5072-11e9-9aee-afe0f889b410.png)


### Notes/TODO
- [x] Clean up code, commented stuff etc
- [x] Continue making sure I didn't break other stuff that uses the single band histogram
- [x] Add more testing instruction steps

## Testing Instructions
* Create a new analysis from the project layers view
* Go to the project analyses page and click the analysis you just created
* Verify that there is an error shown on the histogram
* Adjust the parameters and verify that the histogram now shows and is accurate
* Verify that you can input numbers outside the range of the histogram to force it to resize
* Verify that you can drag the breakpoints, and they will resize to retain a 10% buffer on either side.
* Verify that the histogram will not shrink smaller than the extent of the data (won't cut off graph)
* Verify that changes that you make in the edit panel persist when you select another analysis and then select the same one again
* Verify that focused / visible project analyses can be toggled on the map layers menu separately 
* Verify that analyses created from layers can be edited in the lab
* Verify that changing the project in the lab etc can be reverted by setting parameters in the project analyses quick edit (WIP) 

Closes #4719
Closes #4811
